### DISCUSSION TITLE
Some questions about Auto Mixed Precision Training /expected scalar type Float but found Half

### DISCUSSION THREAD

**Discussion by xxw11** [hpcaitech/ColossalAI#241] [Community | General]
1.当我使用如下配置时会发生错误
fp16 = dict(
    mode=AMP_TYPE.NAIVE
)

![image](https://user-images.githubusercontent.com/55141908/154451349-75a8d13f-443a-4702-b72f-ed7437a17d35.png)
但是我如果切换fp16的类型为AMP_TYPE.TORCH将会正常运行，
`
fp16 = dict(
    mode=AMP_TYPE.TORCH
)
`
我不知道这可能是哪方面的原因？

2.似乎fp16和zero不能一起使用，
我会报如下错误
`It is not allowed to set fp16 and zero configuration in your config file at the same time`
但是当我注释掉关于fp16的设置会出现如下错误
![image](https://user-images.githubusercontent.com/55141908/154455828-213fe1cf-4228-4bd0-a061-50c2c1814263.png)

下面是我的部分设置
`
BATCH_SIZE = 8
SEQ_LEN = 2048
NUM_EPOCHS = 50
TENSOR_PARALLEL = 4

zero = dict(
    level=2,
    dynamic_loss_scale=True,
    overlap_comm=True,
    clip_grad=1.0,
    cpu_offload=False,

)

gradient_accumulation = 5

optimizer = dict(
    type=SGD,
    lr=0.00015,
    weight_decay=1e-2,
)

loss = dict(
    type=GPTLMLoss,
)

model = dict(
    type=gpt2_Y,
    checkpoint=True,
)

parallel = dict(
    pipeline=1,
    tensor=dict(size=TENSOR_PARALLEL, mode='2d'),
)
`

**Comment by ver217** [comment]
Hi, we are trying to reproduce your problem. Could you tell the exact version of your `colossalai`?

**Reply by xxw11** [reply]
colossal version 0.0.1

**Reply by ver217** [reply]
Hi, we have fixed this bug since the `0.0.2` version. Could you give it a try? You can install from `PYPI` or install from source using the latest version.

**Reply by xxw11** [reply]
Thank you so much. Now I can use AMP_TYPE.NAIVE successfully.

**Reply by xxw11** [reply]
Dear developer:
But I still can't use ZeRO 
![image](https://user-images.githubusercontent.com/55141908/154691178-777f22e0-f553-4553-8e0d-da0f988b1c96.png)

**Comment by xxw11** [comment]
Dear developer:
But I still can't use ZeRO 
![image](https://user-images.githubusercontent.com/55141908/154691178-777f22e0-f553-4553-8e0d-da0f988b1c96.png)

**Reply by xxw11** [reply]
Any help would be much appreciated!

**Reply by FrankLeeeee** [reply]
Hi, I suspect it is because of gradient accumulation. Have you tried with grad accumulation = 1?

**Reply by xxw11** [reply]
FrankLeeeee, you are so nice, friendly and helpful. 
After I tried, I found you are right. But I don't understand. Is this a feature?

**Reply by FrankLeeeee** [reply]
ZeRO currently does not work with gradient accumulation well. We are trying to reconstruct zero for better speed and compatibility. We will release this in the next version.

**Reply by FrankLeeeee** [reply]
I have created an issue in #247 , and we will update our progress accordingly. You can keep an eye on this issue if you want to :)

**Reply by xxw11** [reply]
ColossalAI is the best HPC&AI framework I've ever used. It is easy but powerful. It's a really great job. I can't wait to see your next version.

**Reply by FrankLeeeee** [reply]
Thanks for your compliment! Really happy to hear that and we will keep working on this 🚀
